File: src/test/resources/anagrams.pl
#!/usr/bin/env perl

use strict;
use warnings;

use List::Util qw(first);

my @ARGV = @ARGV;

sub load_words {
    my ($file_path) = @_;
    open my $fh, '<', $file_path or die "Cannot open file '$file_path': $!";
    my @words = map { chomp; $_ } <$fh>;
    close $fh;
    return @words;
}

sub find_anagrams {
    my (@words) = @_;
    my %anagrams;
    foreach my $word (@words) {
        my $sorted_word = join("", sort split("", $word));
        push @{$anagrams{$sorted_word}}, $word;
    }
    return %anagrams;
}

sub print_anagrams {
    my (%anagrams) = @_;
    foreach my $sorted_word (keys %anagrams) {
        my @anagram_list = @{$anagrams{$sorted_word}};
        if (@anagram_list > 1) {
            print join(", ", @anagram_list), "\n";
        }
    }
}

if (@ARGV != 1) {
    die "Usage: $0 anagrams.txt\n";
}

my $file_path = shift @ARGV;
my @words = load_words($file_path);
my %anagrams = find_anagrams(@words);
print_anagrams(%anagrams);

File: src/test/resources/anagrams.py
#!/usr/bin/env python3

import sys
from collections import defaultdict


def load_words(file_path):
    with open(file_path, 'r') as file:
        return file.read().splitlines()


def find_anagrams(words):
    anagrams = defaultdict(list)
    for word in words:
        sorted_word = ''.join(sorted(word))
        anagrams[sorted_word].append(word)
    return anagrams


def print_anagrams(anagrams):
    for sorted_word, anagram_list in anagrams.items():
        if len(anagram_list) > 1:
            print(', '.join(anagram_list))


def main():
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} anagrams.txt")
        sys.exit(1)

    file_path = sys.argv[1]
    words = load_words(file_path)
    anagrams = find_anagrams(words)
    print_anagrams(anagrams)


if __name__ == "__main__":
    main()
